### PR TITLE
Add a README.md

### PR DESCRIPTION
**Pull request by jimklimov**
As suggested in https://github.com/jenkinsci/jenkins/pull/2776 a description of this project and recently added improvements is introduced here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# A bit of turbulent history
+
+Since the ZFS filesystem, data integrity and volume management subsystem was
+introduced in the Sun Solaris 10 operating system over a decade ago (in 2005),
+the main interaction API was to use command-line tools. Also an OS-internal
+native library `libzfs.so` was available, but its purpose was interfacing
+those tools to the OS kernel. In absence of other ABIs, a few projects still
+risked to link against this "uncommitted" library -- including the `libzfs.jar`
+wrapper employed by Jenkins CI on matching platforms for tasks like snapshot
```
Comment (oleg-nenashev) [inline README.md]: The correct name is "Jenkins", not "Jenkins CI". https://jenkins.io/press/

File: README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+While the updated JAR tries its best to guess the correct set for a given
+host OS, it includes a way for the end-user (sysadmin) to enforce particular
+implementation for each such function, using environment variables or java
+properties -- and since such settings can be passed from environment outside
+the Jenkins web-application, they would survive eventual upgrades of the
+`jenkins.war`). Also this technique is expandable, to uniformly handle more
+such cases as the future comes down upon us and the `libzfs.jar` sources
+have to be updated again :)
+
+Note that there is more work possible in this area, such as expanding Jenkins
```
Comment (oleg-nenashev) [inline README.md]: There are some usages of the lib outside Jenkins: https://github.com/search?l=Maven+POM&p=3&q=libzfs&type=Code&utf8=%E2%9C%93 , so maybe it makes sense to somehow abstract from it. In this sentence Jenkins could be just removed.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+release of the single OS.
+
+As more time had passed, the Sun OpenSolaris project and later illumos and
+numerous open-source distributions had splintered off from Solaris (which,
+as a brand, went along its own proprietary path and versioning under Oracle
+stewardship).
+
+Also the ZFS technologies were adopted into multiple other operating system
+kernels (including several BSD's, Linux and MacOS efforts), cross-pollinating
+under the foundations of OpenZFS project. The `libzfs.so` library is still
```
Comment (oleg-nenashev) [inline README.md]: hyperlinks would not hurt

**Comment by jimklimov** [comment]
oops, forgot to push the fixes done earlier by comments from @oleg-nenashev

**Pull request merged**